packages/openstef-beam/src/openstef_beam/backtesting/backtest_event_generator.py

```diff
@@ -128,7 +128,9 @@ def _train_iterator(self) -> Iterator[BacktestEvent]:
             if horizon_end > end_time:
                 break
 
-            training_start = current_time - self.forecaster_config.training_context_length
+            training_start = max(
+                current_time - self.forecaster_config.training_context_length, min(self.index).to_pydatetime()
+            )
```

Comment:
```suggestion
            training_start = max(
                current_time - self.forecaster_config.training_context_length,
                cast("pd.Series[pd.Timestamp]", self.index.min()).to_pydatetime()
            )
```

Using python min iterates over the whole series using python. Using pandas min, offloads it to internal C code which is faster. The difference will probably not be noticed, but it's probably a better practise.

The cast solves some type quirks, but might not be needed soon.